File: app/src/main/java/tech/salroid/filmy/syncs/FilmyAuthenticatorService.java
package tech.salroid.filmy.syncs;

import android.app.Service;
import android.content.Intent;
import android.os.IBinder;
/*
 * Filmy Application for Android
 * Copyright (c) 2016 Ramankit Singh (http://github.com/webianks).
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * The service which allows the sync adapter framework to access the authenticator.
 */
public class FilmyAuthenticatorService extends Service {
    // Instance field that stores the authenticator object
    private FilmyAuthenticator mAuthenticator;

    @Override
    public void onCreate() {
        // Create a new authenticator object
        mAuthenticator = new FilmyAuthenticator(this);
    }

    /*
     * When the system binds to this Service to make the RPC call
     * return the authenticator's IBinder.
     */
    @Override
    public IBinder onBind(Intent intent) {
        return mAuthenticator.getIBinder();
    }
}

File: app/src/main/java/tech/salroid/filmy/syncs/FilmyAuthenticatorService.kt
package tech.salroid.filmy.syncs

import android.app.Service
import android.content.Intent
import android.os.IBinder

/**
 * The service which allows the sync adapter framework to access the authenticator.
 */
class FilmyAuthenticatorService : Service() {

    // Instance field that stores the authenticator object
    private var mAuthenticator: FilmyAuthenticator? = null

    override fun onCreate() {
        // Create a new authenticator object
        mAuthenticator = FilmyAuthenticator(this)
    }

    /*
     * When the system binds to this Service to make the RPC call
     * return the authenticator's IBinder.
     */
    override fun onBind(intent: Intent): IBinder? {
        return mAuthenticator!!.iBinder
    }
}